main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -378,22 +396,33 @@ object Parser2 {
     * Means "if (" with no comment in-between "if" and "(".
     */
   private def expect(kind: TokenKind)(implicit s: State): Unit = {
-    if (!eat(kind)) {
-      // TODO: if the actual kind was a comment, we should produce a "Invalid comment" error instead.
-      // Better still, if it is a DocComment we can say "DocComments are allowed above xyz".
-      val error = ParseError(s"Expected $kind, but found ${nth(0)}", SyntacticContext.Unknown, currentSourceLocation())
-      advanceWithError(error)
+    if (eat(kind)) {
+      return
     }
+
+    val mark = open()
+    val error = nth(0) match {
+      case TokenKind.CommentLine | TokenKind.CommentBlock => ParseError(s"Invalid comment", SyntacticContext.Unknown, previousSourceLocation())
```

Comment:
We try to avoid or patterns because they can be deceptive to read/scan.

Reply:
Would you rather I added a branch for `CommentLine` and `CommentBlock` with identical bodies?

Reply:
Yes